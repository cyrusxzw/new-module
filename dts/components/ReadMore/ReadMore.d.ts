export function ReadMore({
  articles,
  className,
}: {
  articles: any;
  className: any;
}): JSX.Element;
export namespace ReadMore {
  namespace propTypes {
    const articles: PropTypes.Requireable<any[]>;
    const className: PropTypes.Requireable<string>;
  }
  namespace defaultProps {
    const articles_1: any;
    export { articles_1 as articles };
    const className_1: any;
    export { className_1 as className };
  }
}
import PropTypes from 'prop-types';
