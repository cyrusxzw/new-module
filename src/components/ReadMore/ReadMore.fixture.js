const ReadMoreFixture = {
  articles: [
    {
      id: 'test_1',
      category: 'Product',
      longTitle: '1. Senorial consectetur adipiscing elit.',
      uri: 'https://www.aesop.com/au/r/careers',
      verticalThumbnail: {
        altText: 'vertical_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/vertical_hub.jpg',
          medium: './assets/images/ContentHubArticle/vertical_hub.jpg',
          small: './assets/images/ContentHubArticle/vertical_hub.jpg',
        },
      },
      horizontalThumbnail: {
        altText: 'horizontal_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          medium: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          small: './assets/images/ContentHubArticle/horizontal_hub.jpg',
        },
      },
      readingTime: '5 minute read',
      onClick: () => {
        console.log('click tracking: test_1'); // eslint-disable-line no-console
      },
    },
    {
      id: 'test_2',
      category: 'Product',
      longTitle:
        '2. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante.',
      uri: 'https://www.aesop.com/au/r/careers',
      verticalThumbnail: {
        altText: 'vertical_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/vertical_hub.jpg',
          medium: './assets/images/ContentHubArticle/vertical_hub.jpg',
          small: './assets/images/ContentHubArticle/vertical_hub.jpg',
        },
      },
      horizontalThumbnail: {
        altText: 'horizontal_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          medium: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          small: './assets/images/ContentHubArticle/horizontal_hub.jpg',
        },
      },
      readingTime: '5 minute read',
      onClick: () => {
        console.log('click tracking: test_2'); // eslint-disable-line no-console
      },
    },
    {
      id: 'test_3',
      category: 'Culture',
      longTitle:
        '3. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare, id dictum ante. Suspendisse et ornare mauris, id dictum ante.',
      uri: 'https://www.aesop.com/au/r/careers',
      verticalThumbnail: {
        altText: 'vertical_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/vertical_hub.jpg',
          medium: './assets/images/ContentHubArticle/vertical_hub.jpg',
          small: './assets/images/ContentHubArticle/vertical_hub.jpg',
        },
      },
      horizontalThumbnail: {
        altText: 'horizontal_hub',
        sizes: {
          large: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          medium: './assets/images/ContentHubArticle/horizontal_hub.jpg',
          small: './assets/images/ContentHubArticle/horizontal_hub.jpg',
        },
      },
      readingTime: '5 minute read',
      onClick: () => {
        console.log('click tracking: test_3'); // eslint-disable-line no-console
      },
    },
  ],
};

export { ReadMoreFixture };
