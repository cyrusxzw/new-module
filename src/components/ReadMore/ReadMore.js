import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { ContentHubArticle } from '~/components/ContentHubArticle/index.ts';
import { Transition } from '~/components/Transition';
import styles from './ReadMore.module.css';

const ReadMore = ({ articles, className }) => {
  const classSet = cx(styles.base, className);
  const readMoreArticleClassSet = cx(styles.readMoreArticle);

  return (
    <Transition isActiveOnMount={true} type="fade">
      <section className={classSet}>
        {articles.map((article) => (
          <ContentHubArticle
            articleRef={article.articleRef}
            category={article.category}
            className={readMoreArticleClassSet}
            dataTestRef={article.id}
            horizontalThumbnail={article.horizontalThumbnail}
            id={article.id}
            isHorizontal={true}
            isInFirstGroup={false}
            isReadMore={true}
            key={article.id}
            longTitle={article.longTitle}
            onClick={article.onClick}
            readingTime={article.readingTime}
            uri={article.uri}
            verticalThumbnail={article.verticalThumbnail}
          />
        ))}
      </section>
    </Transition>
  );
};

ReadMore.propTypes = {
  articles: PropTypes.array,
  className: PropTypes.string,
};

ReadMore.defaultProps = {
  articles: undefined,
  className: undefined,
};

export { ReadMore };
