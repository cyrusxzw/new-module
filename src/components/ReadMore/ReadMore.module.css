@import 'styles/index';

.base {
  @media (--viewport-md-only) {
    width: calc((100% - 34px * 2) * 4 / 6);
    margin: 0 auto;
  }

  @media (--viewport-lg) {
    display: flex;
    max-width: 1440px;
    flex-direction: row;
    justify-content: space-around;
    padding-right: 20px;
    padding-left: 20px;
    margin: 0 auto;
  }
}

.readMoreArticle {
  flex: 1;
  @media (--viewport-xs-sm-only) {
    padding-right: 20px;
    padding-bottom: 50px;
    padding-left: 20px;
  }

  @media (--viewport-md-only) {
    padding-bottom: 85px;
  }
}
