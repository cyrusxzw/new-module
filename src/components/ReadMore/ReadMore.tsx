import React from 'react';
import cx from 'classnames';
import { ContentHubArticle } from '~/components/ContentHubArticle';
import { Transition } from '~/components/Transition';
import styles from './ReadMore.module.css';
import type { ReadMoreType } from './ReadMore.types';

const ReadMore: ReadMoreType = ({ articles, className }) => {
  const classSet = cx(styles.base, className);
  const readMoreArticleClassSet = cx(styles.readMoreArticle);

  return (
    <Transition isActiveOnMount={true} type="fade">
      <section className={classSet}>
        {articles.map((article) => (
          <ContentHubArticle
            articleRef={article.articleRef}
            category={article.category}
            className={readMoreArticleClassSet}
            dataTestRef={article.id}
            horizontalThumbnail={article.horizontalThumbnail}
            id={article.id}
            isHorizontal={true}
            isInFirstGroup={false}
            isReadMore={true}
            key={article.id}
            longTitle={article.longTitle}
            onClick={article.onClick}
            readingTime={article.readingTime}
            uri={article.uri}
            verticalThumbnail={article.verticalThumbnail}
          />
        ))}
      </section>
    </Transition>
  );
};

export { ReadMore };
