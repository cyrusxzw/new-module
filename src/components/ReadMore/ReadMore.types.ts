import type { ComponentWithoutChildren } from '~/types';
import type { Article } from '~/components/ContentHubArticle/ContentHubArticle.types';

type ReadMoreProps = {
  articles: Article[];
  className?: string;
};

type ReadMoreType = ComponentWithoutChildren<ReadMoreProps>;

export { ReadMoreType };
