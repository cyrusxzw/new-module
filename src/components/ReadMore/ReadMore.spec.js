import React from 'react';
import { render } from '@testing-library/react';
import { ReadMore } from './ReadMore';
import { ReadMoreFixture } from './ReadMore.fixture';

describe('<ReadMore />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <ReadMore articles={ReadMoreFixture.articles} />,
    );

    expect(container).toMatchSnapshot();
  });
});
