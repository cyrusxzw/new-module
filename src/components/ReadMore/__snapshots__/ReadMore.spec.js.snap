// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadMore /> should render base component correctly 1`] = `
<div>
  <section
    class="base fadeEnter fadeEnterActive"
  >
    <div
      class="base readMoreArticle fadeEnter fadeEnterActive"
      id="test_1"
    >
      <a
        class="base left dark image"
        data-test-ref="test_1_READMORE_THUMBNAIL"
        href="https://www.aesop.com/au/r/careers"
        target="_self"
        title="1. Senorial consectetur adipiscing elit."
      >
        <div>
          <picture
            class="base picture dark"
          >
            <source
              media="(min-width: 1025px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 640px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 0px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <img
              alt="horizontal_hub"
              loading="lazy"
              src="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
          </picture>
        </div>
      </a>
      <div
        class="category readMoreCategory"
      >
        Product
      </div>
      <div>
        <a
          class="base left dark title readMoreTitle"
          data-test-ref="test_1_TITLE"
          href="https://www.aesop.com/au/r/careers"
          target="_self"
          title="1. Senorial consectetur adipiscing elit."
        >
          <h1>
            <span>
              1. Senorial consectetur adipiscing elit.
            </span>
          </h1>
        </a>
      </div>
      <div
        class="note readMoreNote"
      >
        <span>
          5 minute read
        </span>
      </div>
    </div>
    <div
      class="base readMoreArticle fadeEnter fadeEnterActive"
      id="test_2"
    >
      <a
        class="base left dark image"
        data-test-ref="test_2_READMORE_THUMBNAIL"
        href="https://www.aesop.com/au/r/careers"
        target="_self"
        title="2. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante."
      >
        <div>
          <picture
            class="base picture dark"
          >
            <source
              media="(min-width: 1025px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 640px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 0px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <img
              alt="horizontal_hub"
              loading="lazy"
              src="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
          </picture>
        </div>
      </a>
      <div
        class="category readMoreCategory"
      >
        Product
      </div>
      <div>
        <a
          class="base left dark title readMoreTitle"
          data-test-ref="test_2_TITLE"
          href="https://www.aesop.com/au/r/careers"
          target="_self"
          title="2. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante."
        >
          <h1>
            <span>
              2. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante.
            </span>
          </h1>
        </a>
      </div>
      <div
        class="note readMoreNote"
      >
        <span>
          5 minute read
        </span>
      </div>
    </div>
    <div
      class="base readMoreArticle fadeEnter fadeEnterActive"
      id="test_3"
    >
      <a
        class="base left dark image"
        data-test-ref="test_3_READMORE_THUMBNAIL"
        href="https://www.aesop.com/au/r/careers"
        target="_self"
        title="3. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare, id dictum ante. Suspendisse et ornare mauris, id dictum ante."
      >
        <div>
          <picture
            class="base picture dark"
          >
            <source
              media="(min-width: 1025px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 640px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <source
              media="(min-width: 0px)"
              srcset="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
            <img
              alt="horizontal_hub"
              loading="lazy"
              src="./assets/images/ContentHubArticle/horizontal_hub.jpg"
            />
          </picture>
        </div>
      </a>
      <div
        class="category readMoreCategory"
      >
        Culture
      </div>
      <div>
        <a
          class="base left dark title readMoreTitle"
          data-test-ref="test_3_TITLE"
          href="https://www.aesop.com/au/r/careers"
          target="_self"
          title="3. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare, id dictum ante. Suspendisse et ornare mauris, id dictum ante."
        >
          <h1>
            <span>
              3. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare, id dictum ante. Suspendisse et ornare mauris, id dictum ante.
            </span>
          </h1>
        </a>
      </div>
      <div
        class="note readMoreNote"
      >
        <span>
          5 minute read
        </span>
      </div>
    </div>
  </section>
</div>
`;
